lxd/metrics/api_rates.go

```diff
@@ -0,0 +1,93 @@
+package metrics
+
+import (
+	"net/url"
+	"sync/atomic"
+
+	"github.com/canonical/lxd/shared/entity"
+)
+
+// RequestStatus represents the a completed request status category
+type requestStatus string
+
+const (
+	serverError requestStatus = "server_error"
```

Comment:
lets use errorServer and errorClient with associated strings of "error_*" then we can do prefix matching for all errors

Reply:
This will also require a slight change in the spec but it is fine for me :D